data_transfer/services/inventory.py

```diff
@@ -18,14 +20,27 @@ def device_history(device_id: str) -> Any:
     return response.json()["data"]
 
 
-def patient_id_by_device_id(device_id: str) -> Optional[str]:
-    values = [i for i in device_history(device_id).values()]
+def record_by_device_id(
+    device_id: str, start_wear: datetime, end_wear: datetime
+) -> Optional[Any]:
+    device_wears = [i for i in device_history(device_id).values()]
 
-    if len(values) == 1:
-        return str(values[0]["patient_id"])
-    else:
-        # this means that there are multiple devices in the history, e.g.
-        # that multiple users may have used the same device. Therefore,
-        # we also need to pass in the wear time range and filter by it.
-        # TODO: check which device is within wear time range
-        return None
+    start_wear = utils.normalise_day(start_wear)
+    end_wear = utils.normalise_day(end_wear)
+
+    for record in device_wears:
+        inventory_start_wear = utils.format_weartime(record["checkout"], "inventory")
+        checkin = record["checkin"] or datetime.now().strftime(
+            utils.FORMATS["inventory"]
+        )
+        inventory_end_wear = utils.format_weartime(checkin, "inventory")
+
+        inventory_start_wear = utils.normalise_day(inventory_start_wear)
+        inventory_end_wear = utils.normalise_day(inventory_end_wear)
+
+        within_start_period = inventory_start_wear <= start_wear <= inventory_end_wear
+        within_end_period = inventory_start_wear <= end_wear <= inventory_end_wear
```

Comment:
I now see you use normalised days for comparison. I agree with your current implementation but must admit this doesn't feel fully accurate. Than again - our use case will never have hit or misses in these type of comparisons based on hours - and even not based on days! So all good here.